crates/legacy/src/service.rs

```diff
@@ -1045,13 +1108,41 @@ impl<Types: NodeType> AcceptsTxnSubmits<Types> for ProxyGlobalState<Types> {
             txns.len(),
             txns.iter().map(|txn| txn.commit()).collect::<Vec<_>>()
         );
+        for txn in txns.clone() {
+            self.global_state
+                .write_arc()
+                .await
+                .set_txn_status(txn.commit(), TransactionStatus::Pending)
+                .await
+                .unwrap();
+        }
```

Comment:
Can we get a write lock before this for loop to avoid acquiring it multiple times, and drop it afterward?

Reply:
Yeah I think as long as we make sure everytime we acquire lock on global_state before acquiring lock on txn_status it should be fine.